### PR TITLE
Refactor docker images and workflows to build and deploy them

### PR DESCRIPTION
**Pull request by andistorm**
## Changes
  * Remove workflow `deploy-build-kit.yaml`. It is integrated into
      `deploy-docker-images.yml`.
  * Refactor deploy-docker-images.yml workflow
      * Convert matrix into multiple jobs, one for each docker image,
      because the images depend on each other, and the matrix strategy
      does not support this.
      * Add `env-setup` job to set up the environment for the other jobs,
      because reusable workflows don't support environment variables.
      * Add input parameter to build deprecated images. Currently, the
      deprecated images are built by default, if not specified otherwise.
      * Use output `one_image_tag_short` as input for the dependent images
  * Refactor build-single-docker-image.yml
      * Rename input `docker_directory` to `directory`. Instead of setting the
      directory that contains all images, it is set now to the directory that
      contains the Dockerfile for the image to build.
      * Add input parameter `docker_file_name` to specify the Dockerfile to use.
      * Add input parameter `platforms` to specify the platforms to build the
      image for. The default is `linux/amd64,linux/arm64` and `linux/arm/v7`.
      * Add input parameter `depends_on_paths` to specify the paths that the
      image depends on. If the paths change, the image is rebuilt.
      * Add input parameter `build_args` to specify the build arguments for the
      image.
      * The name of the deployed image is now set by the input parameter
      `image_name`: `ghcr.io/everest/<image_name>`. The directory name is
      independent of the image name.
      * Add output parameter `rebuild` to indicate if the image was rebuilt.
      * Add output parameter `image_tag_short` to pass the image tag to the
      dependent images.
      * Add output parameter `image_tag_long` to pass the full image tag to the
      dependent images.
      * Add output parameter `image_tags` to pass all full image tags.
      * Fix rebuild conditions
  * Update README.md
  * Deprecate `ghcr.io/everest/everest-clang-format` image. Its features are now included in 
      `ghcr.io/everest/everest-ci/build-env-base` The old image is deprecated, but still available and deployed.
  * Merge `ghcr.io/everest/build-kit-alpine` and
      `ghcr.io/everest/build-kit-debian` into `ghcr.io/everest/everest-ci/build-kit-base`.
      The old images are deprecated, but still available and deployed.
  * Add `ghcr.io/everest/everest-ci/run-env-base` image
  * Add `ghcr.io/everest/everest-ci/build-env-base` image
  * Add `ghcr.io/everest/everest-ci/dev-env-base` image
  * Add `run-clang-format.py` script to `build-env-base` image
  * Update github action `run-glang-format` to use new `build-env-base` image

## Related
* https://github.com/EVerest/EVerest/issues/178
* https://github.com/EVerest/everest-core/pull/770
* https://github.com/EVerest/everest-utils/pull/152
* https://github.com/EVerest/everest-dev-environment/pull/45
* https://github.com/EVerest/EVerest/pull/183

### CODE REVIEW

File: github-actions/run-clang-format/action.yaml [hunk 1]
```diff
@@ -19,8 +19,9 @@ inputs:
     default: 'always'
 runs:
   using: 'docker'
-  image: 'docker://ghcr.io/everest/everest-clang-format:v1.1.0'
+  image: 'docker://ghcr.io/everest/everest-ci/build-env-base:refactor-docker-images' # LTODO: Update this
```
Comment (andistorm) [inline github-actions/run-clang-format/action.yaml:22]:
```suggestion
  image: 'docker://ghcr.io/everest/everest-ci/build-env-base:v1.3.0
```

TODO after merge: release `v1.3.0`

File: docker/images/build-env-base/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+RUN apt clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install additional Python packages
+RUN python3 -m pip install --break-system-packages \
+    environs==11.0.0 \
+    aiofile==3.8.8 \
+    py4j==0.10.9.7 \
+    gcovr==7.2
```
Comment (hikinggrass) [inline docker/images/build-env-base/Dockerfile:68]: some of these (eg. aiofile and netifaces) are runtime dependencies of the PyEvJosev module, not sure they belong in the build base

File: docker/images/build-kit-base/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ARG BASE_IMAGE_TAG=latest
+FROM ghcr.io/everest/everest-ci/build-env-base:${BASE_IMAGE_TAG}
+
+ENV WORKSPACE_PATH=/workspace
+ENV ASSETS_PATH=/assets
+ARG EXT_MOUNT=/ext
+ENV EXT_MOUNT=$EXT_MOUNT
+
+RUN mkdir $ASSETS_PATH
+COPY maven-settings.xml $ASSETS_PATH/
```
Comment (hikinggrass) [inline docker/images/build-kit-base/Dockerfile:11]: probably not needed anymore, see above

File: docker/images/build-kit-base/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+setup_cache(){
+    export CPM_SOURCE_CACHE=$EXT_MOUNT/cache/cpm
+    export CCACHE_DIR=$EXT_MOUNT/cache/ccache
+
+    maven_cache_dir=$EXT_MOUNT/cache/m2-repo
+
+    # we copy the settings file each time in case the $HOME folder changes
+    mkdir -p $HOME/.m2 || true
+    cp $ASSETS_PATH/maven-settings.xml $HOME/.m2/settings.xml
+    sed -i 's@MAVEN_CACHE_REPOSITORY@'"$maven_cache_dir"'@' $HOME/.m2/settings.xml
```
Comment (hikinggrass) [inline docker/images/build-kit-base/entrypoint.sh:12]: probably not needed anymore, see above

File: docker/images/build-kit-base/maven-settings.xml
Comment (hikinggrass) [inline docker/images/build-kit-base/maven-settings.xml:1]: probably not needed anymore, see above

File: docker/deprecated-images/build-kit/maven-settings.xml
[the remark this answers is not in the document]
Comment (andistorm) [reply docker/deprecated-images/build-kit/maven-settings.xml:1]: I remove it for the new images

**Review by hikinggrass — APPROVED** [review]

**Pull request merged**